controllers/grid/users/stageParticipant/form/AddParticipantForm.inc.php

```diff
@@ -71,17 +118,47 @@ function fetch($request, $template = null, $display = false) {
 		// assigned the first element as selected
 		$keys = array_keys($userGroupOptions);
 		$templateMgr->assign('selectedUserGroupId', array_shift($keys));
-		// assign all user group IDs with ROLE_ID_MANAGER or ROLE_ID_SUB_EDITOR
-		$managerGroupIds = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_MANAGER, $request->getContext()->getId());
-		$subEditorGroupIds = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_SUB_EDITOR, $request->getContext()->getId());
-		$possibleRecommendOnlyUserGroupIds = array_merge($managerGroupIds, $subEditorGroupIds);
-		$templateMgr->assign('possibleRecommendOnlyUserGroupIds', $possibleRecommendOnlyUserGroupIds);
+
+		$templateMgr->assign('possibleRecommendOnlyUserGroupIds', $this->_possibleRecommendOnlyUserGroupIds);
 		// assign user group IDs with recommendOnly option set
 		$templateMgr->assign('recommendOnlyUserGroupIds', $userGroupDao->getRecommendOnlyGroupIds($request->getContext()->getId()));
 
+		$templateMgr->assign('notPossibleEditSubmissionMetadataPermissionChange', $this->_managerGroupIds);
+		$templateMgr->assign('permitMetadataEditUserGroupIds', $userGroupDao->getPermitMetadataEditGroupIds($request->getContext()->getId()));
+
 		// assign the vars required for the request
 		$templateMgr->assign('submissionId', $this->getSubmission()->getId());
 
+		$templateMgr->assign('userGroupId', '');
+		$templateMgr->assign('userIdSelected', '');
+
+		if ($this->_assignmentId) {
+			/** @var $stageAssignmentDao StageAssignmentDAO */
+			$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
+
+			/** @var $stageAssignment StageAssignment */
+			$stageAssignment = $stageAssignmentDao->getById($this->_assignmentId);
+
+			$userDao = DAORegistry::getDAO('UserDAO');
+			/** @var $currentUser User */
+			$currentUser = $userDao->getById($stageAssignment->getUserId());
+
+			$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
+			/** @var $userGroup UserGroup */
+			$userGroup = $userGroupDao->getById($stageAssignment->getUserGroupId());
+
+			$templateMgr->assign('assignmentId', $this->_assignmentId);
```

Comment:
Use this syntax for long sets of assignments:
```
$templateMgr->assign(array(
    ...
));
```

Reply:
Done